src/pages/index.js

```diff
@@ -14,81 +16,190 @@ const addCardAddButton = document.querySelector(selectors.cardAddButton);
-  const card = new Card(item, selectors.cardTemplate, handleImageClick);
-  return card.generateCard();
-}
+// Edit avatar popup
+const avatarPopup = new PopupWithForm(selectors.avatarPopup, handleEditAvatar);
+avatarPopup.setEventListeners();
 
-// ! ||--------------------------------------------------------------------------------||
-// ! ||                                 Event Listeners                                ||
-// ! ||--------------------------------------------------------------------------------||
+// Event Listeners
 addCardAddButton.addEventListener("click", () => {
   newCardPopup.open();
   cardFormValidator.toggleButtonState();
 });
 
 updateProfileButton.addEventListener("click", () => {
   const userData = userInfo.getUserInfo();
-  profileHeadingInput.value = userData.userName;
-  profileDescriptionInput.value = userData.userDescription;
+  profileHeadingInput.value = userData.name;
+  profileDescriptionInput.value = userData.about;
   editProfileFormValidator.resetValidation();
   updateProfilePopup.open();
 });
 
-// ! ||--------------------------------------------------------------------------------||
-// ! ||                                 Event Handlers                                 ||
-// ! ||--------------------------------------------------------------------------------||
-function handleAddCardFormSubmit(data) {
-  const card = renderCard(data);
-  cardSection.addItem(card);
+updateAvatarButton.addEventListener("click", () => {
+  avatarPopup.open();
+});
+
+// Functions
+function handleValidation(form) {
+  form.enableValidation();
+}
+
+function handleAddCardFormSubmit(cardData) {
+  newCardPopup.showButtonProgress(true);
+  api.addNewCard(cardData.name, cardData.link)
+    .then((res) => {
+      const card = renderCard(res);
+      cardSection.addItem(card);
+      newCardPopup.reset();
+      newCardPopup.close();
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    })
+    .finally(() => {
+      newCardPopup.showButtonProgress(false);
+    });
 }
 
 function handleImageClick(name, link) {
   newImagePopup.open(name, link);
 }
 
 function handleProfileSubmit(userData) {
-  userInfo.setUserInfo(userData);
-}
\ No newline at end of file
+  updateProfilePopup.showButtonProgress(true);
+  api.updateUserInfo(userData.name, userData.about)
+    .then((user) => {
+      userInfo.setUserInfo(user);
+      updateProfilePopup.reset();
+      updateProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    })
+    .finally(() => {
+      updateProfilePopup.showButtonProgress(false);
+    });
+}
+
+function handleDeleteCard(cardData) {
+  deletePopup.open();
+  deletePopup.setSubmitAction(() => {
+    deletePopup.showButtonProgress(true);
+    api.deleteCard(cardData._id)
+      .then(() => {
+        cardData.handleRemoveCard();
+        deletePopup.close();
+      })
+      .catch((err) => {
+        console.error(`Error: ${err}`);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors